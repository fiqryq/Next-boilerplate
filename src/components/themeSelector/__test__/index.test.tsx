import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import ThemeSelector from '../index';

const describeErrorComponent = '<ThemeSelector /> test case';

describe(describeErrorComponent, () => {
  const { container } = render(<ThemeSelector />);
  test(`Render ${describeErrorComponent} correctly`, () => {
    expect(container).toMatchSnapshot();
  });
});
