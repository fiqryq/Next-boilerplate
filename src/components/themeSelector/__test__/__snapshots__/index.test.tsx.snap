// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeSelector /> test case Render <ThemeSelector /> test case correctly 1`] = `
<div>
  <div
    class="relative"
    data-headlessui-state=""
  >
    <label
      class="sr-only"
      data-headlessui-state=""
      id="headlessui-listbox-label-:r0:"
    >
      Theme
    </label>
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="headlessui-listbox-label-:r0: headlessui-listbox-button-:r1:"
      class="flex h-6 w-6 items-center justify-center rounded-lg shadow-md shadow-black/5 ring-1 ring-black/5 dark:bg-slate-700 dark:ring-inset dark:ring-white/5"
      data-headlessui-state=""
      id="headlessui-listbox-button-:r1:"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="hidden h-4 w-4 fill-sky-400 [[data-theme=light]_&]:block"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M7 1a1 1 0 0 1 2 0v1a1 1 0 1 1-2 0V1Zm4 7a3 3 0 1 1-6 0 3 3 0 0 1 6 0Zm2.657-5.657a1 1 0 0 0-1.414 0l-.707.707a1 1 0 0 0 1.414 1.414l.707-.707a1 1 0 0 0 0-1.414Zm-1.415 11.313-.707-.707a1 1 0 0 1 1.415-1.415l.707.708a1 1 0 0 1-1.415 1.414ZM16 7.999a1 1 0 0 0-1-1h-1a1 1 0 1 0 0 2h1a1 1 0 0 0 1-1ZM7 14a1 1 0 1 1 2 0v1a1 1 0 1 1-2 0v-1Zm-2.536-2.464a1 1 0 0 0-1.414 0l-.707.707a1 1 0 0 0 1.414 1.414l.707-.707a1 1 0 0 0 0-1.414Zm0-8.486A1 1 0 0 1 3.05 4.464l-.707-.707a1 1 0 0 1 1.414-1.414l.707.707ZM3 8a1 1 0 0 0-1-1H1a1 1 0 0 0 0 2h1a1 1 0 0 0 1-1Z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="hidden h-4 w-4 fill-sky-400 [[data-theme=dark]_&]:block"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M7.23 3.333C7.757 2.905 7.68 2 7 2a6 6 0 1 0 0 12c.68 0 .758-.905.23-1.332A5.989 5.989 0 0 1 5 8c0-1.885.87-3.568 2.23-4.668ZM12 5a1 1 0 0 1 1 1 1 1 0 0 0 1 1 1 1 0 1 1 0 2 1 1 0 0 0-1 1 1 1 0 1 1-2 0 1 1 0 0 0-1-1 1 1 0 1 1 0-2 1 1 0 0 0 1-1 1 1 0 0 1 1-1Z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="hidden h-4 w-4 fill-slate-400 [:not(.dark)[data-theme=system]_&]:block"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M7 1a1 1 0 0 1 2 0v1a1 1 0 1 1-2 0V1Zm4 7a3 3 0 1 1-6 0 3 3 0 0 1 6 0Zm2.657-5.657a1 1 0 0 0-1.414 0l-.707.707a1 1 0 0 0 1.414 1.414l.707-.707a1 1 0 0 0 0-1.414Zm-1.415 11.313-.707-.707a1 1 0 0 1 1.415-1.415l.707.708a1 1 0 0 1-1.415 1.414ZM16 7.999a1 1 0 0 0-1-1h-1a1 1 0 1 0 0 2h1a1 1 0 0 0 1-1ZM7 14a1 1 0 1 1 2 0v1a1 1 0 1 1-2 0v-1Zm-2.536-2.464a1 1 0 0 0-1.414 0l-.707.707a1 1 0 0 0 1.414 1.414l.707-.707a1 1 0 0 0 0-1.414Zm0-8.486A1 1 0 0 1 3.05 4.464l-.707-.707a1 1 0 0 1 1.414-1.414l.707.707ZM3 8a1 1 0 0 0-1-1H1a1 1 0 0 0 0 2h1a1 1 0 0 0 1-1Z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="hidden h-4 w-4 fill-slate-400 [.dark[data-theme=system]_&]:block"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M7.23 3.333C7.757 2.905 7.68 2 7 2a6 6 0 1 0 0 12c.68 0 .758-.905.23-1.332A5.989 5.989 0 0 1 5 8c0-1.885.87-3.568 2.23-4.668ZM12 5a1 1 0 0 1 1 1 1 1 0 0 0 1 1 1 1 0 1 1 0 2 1 1 0 0 0-1 1 1 1 0 1 1-2 0 1 1 0 0 0-1-1 1 1 0 1 1 0-2 1 1 0 0 0 1-1 1 1 0 0 1 1-1Z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
